workspaces/src/types.rs

```diff
@@ -2,11 +2,30 @@
 /// and internal libraries like near-jsonrpc-client requires specific versions
 /// of these types which shouldn't be exposed either.
 use std::convert::TryFrom;
+use std::fmt;
 use std::path::Path;
 
+/// Error module pertaining to all the error types exposed by workspaces. This
+/// is not an exhaustive list of errors and more can be added in the future.
+pub mod errors {
+    pub use near_primitives::errors::{
+        ActionError, ActionErrorKind, InvalidAccessKeyError, InvalidTxError, TxExecutionError,
+    };
+}
+
+/// Status module exposing types where we can view the status of a transaction's
+/// outcome.
+pub mod status {
+    pub use near_primitives::views::{ExecutionStatusView, FinalExecutionStatus};
+}
```

Comment:
I'm skeptical it's the best long-term plan to expose all of these types to the API, seems like it would just be a maintenance pain to do so. I don't have a great grasp on why this is needed just yet

Reply:
For context, this would be needed if a user wants to grab data from the status of an outcome and match it to a specific error type to do whatever they wish with it. We already offer a `failures()` API, but that might not suit all cases.

The alternative could just be forcing users to use `near_primitives::errors::*` instead, but think we had the discussion of that not being so great either. The error types are pretty big, but I'd imagine exposing them to not hurt us so much since eventually `near_primitives` 1.0 should be released. @miraclx do you have any opinions on this?

Reply:
> This might be hard to maintain in the future as the near-primitives crate gets updated. So I think it's best if we just expose it as-is for now since this is just a view type where the users can grab the data.

This sounds good to me. Let's not expose the users directly to a crate, whose API is certain to change a lot very soon. Instead, by providing those types from within workspaces, we abstract over them. At least until the new crate is ready. At that point, if we decide it's the better way forward, we can drop them from workspaces and users can migrate to using that directly.

Reply:
@austinabell The errors are now abstracted behind anyhow ones and are no longer exposed. Let me know what you think